### ISSUE TITLE
ExperientialMemoryPolicy sums float32 action mass without overflow protection, incorrectly abstaining on valid retrievals

### ISSUE THREAD

**Issue by Svector-anu** [SlopDotCash/asi#314]
`ExperientialMemoryPolicy._interpret_jit` directly summed the retrieved float32 categorical action masses via `jnp.sum(safe_finite_mass)`. A valid, finite, nonnegative mass vector such as `[float32_max, float32_max, float32_max]` overflows that sum to `inf`. The policy then set `action_mass_valid=False` and abstained with action `-1`, even though the memory accepted the retrieval and the documented categorical distribution is exactly `[1/3, 1/3, 1/3]` with lowest-index action `0`.

confirmed directly through the real public path (`ExperientialMemory.write` → `.query` → `ExperientialMemoryPolicy.propose`):
```text
wrote True
retrieval.accepted True
retrieval.action [3.4028235e+38 3.4028235e+38 3.4028235e+38]
total_action_mass inf
action_mass_valid False
available False
action -1
normalized [0. 0. 0.]
```

reachability: `ExperientialMemoryPolicy` is exported from both `alberta_framework.core` and the root `alberta_framework` package. `PrototypeAgent` constructs one when experiential memory is enabled (`alberta_framework/core/prototype_agent.py:2321`); its public `scan_transitions` path reaches `policy.propose(...)` with caller-supplied finite float32 action-mass vectors that are not hardcoded to a safe magnitude — the overflow case is reachable with real inputs, not just a synthetic edge.

fix incoming: scale mass by its own max via `frexp`/`ldexp` before summing/normalizing (avoids the intermediate overflow entirely and is scale-invariant, so `normalized_action_mass` is correct regardless of magnitude), and report a finite-saturated float32 total instead of `inf` when the true sum exceeds the representable diagnostic range.
